src/main/java/com/exactpro/th2/conn/MicroserviceMain.java

```diff
@@ -411,20 +412,52 @@ public static IServiceProxy loadService(IServiceFactory serviceFactory,
-                    settings.setDictionary(sfDictionaryType, uri);
-                }
-            }
+            loadDictionaries(serviceFactory, commonFactory, configuration, settings);
 
             return service;
         } catch (ConversionException | ServiceFactoryException e) {
             throw new RuntimeException(String.format("Could not load service '%s'", configuration.getName()), e);
         }
     }
 
+    private static void loadDictionaries(IServiceFactory serviceFactory, CommonFactory commonFactory, ConnectivityConfiguration configuration, ISettingsProxy settings) throws IOException, ServiceFactoryException {
+        var dictionariesToAliasMap = configuration.getDictionariesToAliasMap();
+        if (dictionariesToAliasMap != null && !dictionariesToAliasMap.isEmpty()) {
+            loadDictionariesByAliases(serviceFactory, commonFactory, settings, dictionariesToAliasMap);
+        } else {
+            loadDictionariesByTypes(serviceFactory, commonFactory, settings);
+        }
+    }
+
+    private static void loadDictionariesByTypes(IServiceFactory serviceFactory, CommonFactory commonFactory, ISettingsProxy settings) throws IOException, ServiceFactoryException {
+        LOGGER.debug("Loading dictionaries by types...");
+        for (DictionaryType sfDictionaryType : settings.getDictionaryTypes()) {
+            var dictionaryType = com.exactpro.th2.common.schema.dictionary.DictionaryType.valueOf(sfDictionaryType.name());
+            try (InputStream stream = commonFactory.readDictionary(dictionaryType)) {
+                SailfishURI uri = serviceFactory.registerDictionary(sfDictionaryType.name(), stream, true);
+                settings.setDictionary(sfDictionaryType, uri);
+            }
+        }
+    }
+
+    private static void loadDictionariesByAliases(IServiceFactory serviceFactory, CommonFactory commonFactory, ISettingsProxy settings, Map<String, String> dictionariesToAliasMap) throws IOException, ServiceFactoryException {
+        Set<String> commonDictionaryAliases = commonFactory.getDictionaryAliases();
+        LOGGER.debug("Loading dictionaries by aliases...");
+        for (Entry<String, String> entry : dictionariesToAliasMap.entrySet()) {
+            DictionaryType sfDictionaryType = DictionaryType.valueOf(entry.getKey());
+            String dictionaryAlias = entry.getValue();
+            var contains = commonDictionaryAliases.contains(dictionaryAlias);
+            if (contains) {
+                try (InputStream inputStream = commonFactory.loadDictionary(dictionaryAlias)) {
+                    SailfishURI uri = serviceFactory.registerDictionary(sfDictionaryType.name(), inputStream, true);
+                    settings.setDictionary(sfDictionaryType, uri);
+                }
+            } else {
+                throw new IllegalArgumentException(String.format("Could not find dictionary with alias '%s'", dictionaryAlias));
+            }
+        }
```

Comment:
Also, conn should check that only required dictionary types are specified in the dictionariesToAliasMap and throw an exception on otherwise.
Each component is responsible for supporting configuration in the actual state.